### PR TITLE
using prefix on joints didn't work with gazebo controllers. Fixed now

### PR DESCRIPTION
**Pull request by ugocupcic**
tested on gazebo hand + gazebo dual hand with different prefixes (moley)

### CODE REVIEW

File: sr_mechanism_controllers/src/sr_controller.cpp [hunk 1]
```diff
@@ -92,16 +92,25 @@ namespace controller {
...
-      std::string j2 = joint_name.substr(0,3) + "2";
-
-      boost::shared_ptr<const urdf::Joint> joint1 = model.getJoint( j1 );
-      boost::shared_ptr<const urdf::Joint> joint2 = model.getJoint( j2 );
 
-      min_ = joint1->limits->lower + joint2->limits->lower;
-      max_ = joint1->limits->upper + joint2->limits->upper;
+    if(joint_name.size() > 3)
+    {
+      if (joint_name[joint_name.size() - 1] == '2' && (joint_name[joint_name.size() -4] == 'F' ||
```
Comment (toliver) [inline sr_mechanism_controllers/src/sr_controller.cpp]: Shouldn't this be like before, trying to find J0? (Thus we don't need to check the finger, as the thumb doesn't have J0)

```
if (joint_name_[joint_name.size() - 1]  ==  '0')
```

Comment (ugocupcic) [inline sr_mechanism_controllers/src/sr_controller.cpp]: I don't mind either way. Kept it the same as the other one as it seemed easier.

Comment (toliver) [inline sr_mechanism_controllers/src/sr_controller.cpp]: But I'm not sure that it has the same effect.

**Comment by guihomework** [comment]
I would like to comment on that.
The simplest the required clear rules on joint naming the better. 
I read you now suppose all the joint names end with the joint number, meaning you will always replace the last number to handle j0's. 
This code suppose now (and before actually) that the joint names are formed like  xx...xxxx[F|M|R|L]xx#
This is currently true with names or simply prefixed names but maybe one wants that to be sure the finger name initial is always included in the joint name. 
Then there are 2 rules to be respected all the time and they should be documented. For instance I have seen Pinky for the little finger, somewhere which might be a problem if somebody starts using  left_PJ0 (left pinky joint 0) for instance.

**Comment by toliver** [comment]
This will do for the time being. We assume the last char is the joint number.
Currently we assume that the joint name will be `prefix + xx + 'J' + N`.

**Pull request merged**